spec/models/reports/patient_days_calculation_spec.rb

```diff
@@ -0,0 +1,220 @@
+        patient_count: patient_count
+      ).stocks_on_hand
+      expect(result.map { |stock_on_hand| stock_on_hand[:protocol_drug].rxnorm_code }).to match_array []
+    end
+
+    it "computes stock as 0 when all stocks are 0" do
+      result = described_class.new(
+        state: state,
+        protocol: protocol,
+        drug_category: drug_category,
+        stocks_by_rxnorm_code: {"329526" => 0, "329528" => 0},
+        patient_count: patient_count
+      ).stocks_on_hand
+      expect(result.map { |stock_on_hand| stock_on_hand[:stock_on_hand] }).to match_array [0, 0]
+    end
+
+    it "errors out when there is no formula for a state" do
+      allow_any_instance_of(described_class).to receive(:patient_days_coefficients).and_return(nil)
+      instance = described_class.new(
+        state: state,
+        protocol: protocol,
+        drug_category: drug_category,
+        stocks_by_rxnorm_code: stocks_by_rxnorm,
+        patient_count: patient_count
+      )
+      expect { instance.stocks_on_hand }.to raise_error(NoMethodError)
+    end
+  end
+
+  describe "#calculate" do
+    before do
+      allow_any_instance_of(described_class).to receive(:patient_days_coefficients).and_return(punjab_drug_stock_config)
+    end
+    it "works correctly when all inputs are present" do
+      result = described_class.new(
+        state: state,
+        protocol: protocol,
+        drug_category: drug_category,
+        stocks_by_rxnorm_code: stocks_by_rxnorm,
+        patient_count: patient_count
+      ).calculate
+      expect(result[:load_coefficient]).to eq 1
+      expect(result[:new_patient_coefficient]).to eq 1.4
+      expect(result[:patient_count]).to eq patient_count
+      expect(result[:estimated_patients]).to eq 14000
+      expect(result[:patient_days]).to eq ((10000 * 1 + 20000 * 2) / 14000).floor
+    end
+
+    it "patient count is nil" do
+      result = described_class.new(
+        state: state,
+        protocol: protocol,
+        drug_category: drug_category,
+        stocks_by_rxnorm_code: stocks_by_rxnorm,
+        patient_count: nil
+      ).calculate
+      expect(result).to eq(patient_days: "error")
+    end
+
+    it "patient count is 0" do
+      result = described_class.new(
+        state: state,
+        protocol: protocol,
+        drug_category: drug_category,
+        stocks_by_rxnorm_code: stocks_by_rxnorm,
+        patient_count: 0
+      ).calculate
+      expect(result).to eq(patient_days: "error")
+    end
+
+    it "returns an error for patient days when there is no formula for a state" do
+      allow_any_instance_of(described_class).to receive(:patient_days_coefficients).and_return(nil)
+      result = described_class.new(
+        state: state,
+        protocol: protocol,
+        drug_category: drug_category,
+        stocks_by_rxnorm_code: stocks_by_rxnorm,
+        patient_count: patient_count
+      ).calculate
+      expect(result).to eq(patient_days: "error")
+    end
+
+    it "sentry" do
```

Comment:
it "reports exceptions"? For a more natural reading :)

Reply:
haha! we wrote this down as a placeholder while listing all tests to write. will fix.

Reply:
Fixed in 3795213aa1a165ebc6ed76e3fb4b333014e240c6.